fastcrypto/src/hash.rs

```diff
@@ -192,3 +194,76 @@ impl HashFunction<32> for Blake3 {
         }
     }
 }
+
+/// The Accumulator is a homomorphic multiset hash function, which hashes arbitrary multisets
+/// of objects such that the hash of the union of two collections is easy to compute from the
+/// hashes of the two collections.
+///
+/// Concretely, each element is mapped to a point on an elliptic curve on which the DL problem
+/// is hard. The accumulator is the sum of all points.
+///
+/// See for more information about the construction and its security: https://arxiv.org/abs/1601.06502.
+#[derive(Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct Accumulator {
+    accumulator: RistrettoPoint,
+}
+
+impl Accumulator {
```

Comment:
Wondering if all of these functions are common between accumulators and we should put in the trait then implement them here. That will help adding more accumulator schemes in the future.

Reply:
I've now added a trait, _MultisetHash_, and I think that made the code more clear. Let me know if you think it's too abstract for our purpose, because I don't know if we plan to actually have multiple implementations.